webdriver-ts/src/timeline.ts

```diff
@@ -1,378 +1,523 @@
+  ts: number;
+  dur?: number;
+  end?: number;
+  mem?: number;
+  pid: number;
+  evt?: any;
+}
 
-  function logEvents(events: Timingresult[], click: Timingresult) {
-    events.forEach(e => {
-      console.log( "event",e.type, `${e.ts-click.ts} - ${e.end-click.ts}`, e.evt);
-    })
-  }
-  
-  export async function computeResultsCPU(fileName: string, warning_logger: (...msg:any) => void = console.log): Promise<CPUDurationResult> {
-    const perfLogEvents = (await fetchEventsFromPerformanceLog(fileName));
-    let events = R.sortBy((e: Timingresult) => e.end)(perfLogEvents);
-  
-    // Find click event. This is the start of the benchmark
-    let clicks = R.filter(type_eq('click'))(events)
-    // Invariant: There must be exactly one click event
-    if (clicks.length !== 1) {
-        console.log("exactly one click event is expected", fileName, events);
-        throw "exactly one click event is expected";
-    }
-    let click = clicks[0];
-    // The PID for the click event. We're dropping all events from other processes.
-    let pid = click.pid;
-    let eventsDuringBenchmark = R.filter((e: Timingresult) => (e.ts > click.end || e.type === 'click'))(events);
-    if (config.LOG_DETAILS) logEvents(eventsDuringBenchmark, click);
-
-    let droppedNonMainProcessCommitEvents = false;
-    let droppedNonMainProcessOtherEvents = false;
-    let eventsOnMainThreadDuringBenchmark = R.filter((e: Timingresult) => e.pid === pid)(eventsDuringBenchmark);
-    if (eventsOnMainThreadDuringBenchmark.length !== eventsDuringBenchmark.length) {
-      let droppedEvents = R.filter((e: Timingresult) => e.pid !== pid)(events);
-      if (R.any((e: Timingresult) => e.type === 'commit')(droppedEvents)) {
-        console.log("INFO: Dropping commit events from other processes", fileName);
-        logEvents(droppedEvents, click);
-        droppedNonMainProcessCommitEvents = true
-      }
-      if (R.any((e: Timingresult) => e.type !== 'commit')(droppedEvents)) {
-        console.log("INFO: Dropping non-commit events from other processes", fileName);
-        logEvents(droppedEvents, click);
-        droppedNonMainProcessOtherEvents = true;
+export function extractRelevantEvents(entries: any[]) {
+  let filteredEvents: Timingresult[] = [];
+  let click_start = 0;
+  let click_end = 0;
+
+  entries.forEach((x) => {
+    let e = x;
+    if (config.LOG_DEBUG) console.log(JSON.stringify(e));
+    if (e.name === "EventDispatch") {
+      if (e.args.data.type === "click") {
+        if (config.LOG_DETAILS) console.log("CLICK ", +e.ts);
+        click_start = +e.ts;
+        click_end = +e.ts + e.dur;
+        filteredEvents.push({
+          type: "click",
+          ts: +e.ts,
+          dur: +e.dur,
+          end: +e.ts + e.dur,
+          pid: e.pid,
+          evt: JSON.stringify(e),
+        });
       }
+    } else if (e.name === "Layout" && e.ph === "X") {
+      if (config.LOG_DETAILS) console.log("Layout", +e.ts, +e.ts + e.dur - click_start);
```

Comment:
Readability was in my opinion MUCH better for the old version.

Reply:
It's a matter of taste, but for example on my laptop 16 inches old code, although in some moments and looks okay, but it is more difficult to read from left to right lines of 120+ characters. Besides, most of this code can be quickly made to look nice by refactoring it a bit, because in this PR only formatting.

Reply:
I just thought it would be better to add prettier everywhere and format the code first before refactoring it later